Require Python 3.5+

jakirkham opened the pull request:
As Python 3.5 is now required, make sure the `python` dependency has this constraint applied.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Partially addresses ( https://github.com/conda-forge/numpydoc-feedstock/issues/13 ).

<!--
Please add any other relevant info below:
-->

Also patching repodata with PR ( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/89 ).

jakirkham committed:
MNT: Re-rendered with conda-build 3.20.2, conda-smithy 3.7.10, and conda-forge-pinning 2020.10.01.19.13.14

jakirkham committed:
Require Python 3.5+

As Python 3.5 is now required, make sure the `python` dependency has
this constraint applied.

jakirkham committed:
Bump build number to 1

Rebuild now that the Python 3.5+ requirement is included.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

jakirkham committed:
Use SPDX compatible `license` metadata

jakirkham committed:
MNT: Re-rendered with conda-build 3.20.2, conda-smithy 3.7.10, and conda-forge-pinning 2020.10.01.19.13.14

Now that `license` is SPDX compatible, re-render to update other
feedstock content accordingly.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.
